xarray/plot/plot.py

```diff
@@ -428,6 +467,9 @@ def line(
     ax = get_axis(figsize, size, aspect, ax)
     xplt, yplt, hueplt, hue_label = _infer_line_data(darray, x, y, hue)
 
+    if not isinstance(darray, GroupBy):
+        darray = darray.squeeze().compute()
```

Comment:
Is this compute necessary? Because the ax.plot uses `xplt_val`, `yplt_val` as inputs and those does not appear to be directly dependent on `darray`?